### ISSUE TITLE
Event Start and Stop Time are in the wrong TZ

### ISSUE THREAD

**Issue by Gradlon** [pretix/pretix#4442]
### Problem and impact

When updating an event Time, the time is offset to the actually selected Tim.
Ex: 
TZ is Europe/Zurich setting the time event start time to 08:30 results in a saved time of 10:30. Which is the UTC offset to the Time in Zurich.
If the event is in winter time, the offset is only 1h.

If you update events several times without changing the time, the event is moved every time you save by 1 or 2 hours.

Pretix server runs inside Docker:
Docker timezone is Europe/Zurich
Image: pretix/standalone:2024.7

### Expected behaviour

Expect the time in the input to be the same as the the time in the output.

### Steps to reproduce

1. Create an event
2. add an event time
3. select a date and start date a time
4. look at the event date and time

### Screenshots

<img width="1185" alt="image" src="https://github.com/user-attachments/assets/ac731178-5ca2-44ac-a177-07956dd304f6">
<img width="1391" alt="image" src="https://github.com/user-attachments/assets/a323b42c-4ca8-4a63-901d-e69c9ef4c9b9">

### Browser (software, desktop or mobile?) and version

Chrome

### Operating system, dependency versions

Macos

### Version

pretix/standalone:2024.7

**Comment by wiffbi** [comment]
Could you double check in the database that the time is really shifted? It could also „just“ be a display issue in the events overview or actually just a timezone mismatch between the events timezone (Europe/Zurich) and your default timezone of the installation or even pretix-control user’s timezone-settings in /control/settings – the latter will be used in the events-list overview, where according to your screenshots there is a time-mismatch. Although there should be a little globe-icon indicating that the event is in a different timezone than the current display timezone. So I am still a bit puzzled.

**Comment by Gradlon** [comment]
@wiffbi
![image](https://github.com/user-attachments/assets/ff21596f-ef4c-45c8-9c96-448919c09b9b)
![image](https://github.com/user-attachments/assets/12746790-d4aa-4aec-9b6b-029706038571)

So if I see this correctly, the DB holds correct Values, but the UI adds 2h to it.

**Comment by raphaelm** [comment]
This is really weird and not something we've seen before in development or production. I'm not really sure what we can do to debug it further.

**Comment by wiffbi** [comment]
The only thing that I noticed as a bit odd is the timezone +02 on the timestamp in the database. AFAIK we save timestamps in UTC – at least in dev mode on sqlite, which does not support timezones. AFAICT django does not change that behaviour when using postgresql (which does have a timezone aware datetime field as an extension to the standard). Maybe that timezone-info addon is something to look for? In my tests on a local sqlite dev-setup, setting the date_from to 2024-09-13 14:00:00+02 through the db-interface and a timezone Europe/Zurich results in the correct behaviour to display 14:00:00 in the pretix-form. When I save the form, the timestamp is converted to non-timezone UTC, which results in 2024-09-13 12:00:00 – which is still correctly displayed as 14:00:00 in the form, as Zurich is UTC+2. Is there maybe some setting for your db that automatically interprets any datetimes to be saved as UTC+2 and adds that +02? And then that +02 is somehow cut off when reading the datetime from DB? Because only 14:00:00 UTC would result in 16:00:00 in for Zurich/UTC+02.

**Comment by raphaelm** [comment]
Yeah, might be that somehow the timezone is set in postgres twice, but why only for this installation?

**Comment by Gradlon** [comment]
@raphaelm where would that be?
I have no cloud why it would be set twice.

**Comment by raphaelm** [comment]
:shrug:  I also don't think it's necessarily wrong that it's saved as +02 in the database since PostgreSQL is technically tz-aware, so it should work, whereas in other databases django must only store UTC since they are not tz-aware. Unfortunatelsy, I have no good ideas here currently without reproducing it.

**Comment by wiffbi** [comment]
This might be a long shot, but could you check the timzone your postgres runs in – `SHOW TIMEZONE;` – and change it to UTC? As Raphael pointed out it should not make a difference, but 🤷

**Comment by Gradlon** [comment]
@wiffbi Yes after reading the comment from @raphaelm that's what I was planing to do.
I will try to do this later that day or tomorrow.

**Comment by Gradlon** [comment]
I changed the timezone to UTC but the problem persists.
I also changed the timezone of the container to UTC, but this did not change anything.

**Comment by Gradlon** [comment]
I update the timezones to UTC of
- the container running Pretix
- The Event host
- The Event
- Postgres
But the Problem is still present.

**Comment by hubcombrussels-collab** [comment]
Hi! 
We're encountering the same issue. 
Also a docker installation and running the latest Pretix Version (2025.8) 

Did you find a solution?

**Comment by Gradlon** [comment]
I solved it somehow.
As far as I remember, it had something to do with TZ on the Postgres container.
Will check as soon as I have time.

**Comment by Gradlon** [comment]
This part of my pretix config

[locale]
default=de
timezone=UTC

And I left the TZ for Postgres to the default (runs in docker).

I am not sure exactly what it was that solved the problem, but it had to do with the time zones.
